c10/core/DispatchKeySet.cpp

```diff
@@ -2,13 +2,71 @@
 
 namespace c10 {
 
+// backend dispatch keys that map to DispatchKey::AutogradOther
+constexpr DispatchKeySet autogradother_backends = DispatchKeySet({
+  DispatchKey::HIP,
+  DispatchKey::FPGA,
+  DispatchKey::MSNPU,
+  DispatchKey::Vulkan,
+  DispatchKey::MKLDNN,
+  DispatchKey::OpenGL,
+  DispatchKey::OpenCL,
+  DispatchKey::IDEEP,
+  DispatchKey::QuantizedCPU,
+  DispatchKey::QuantizedCUDA,
+  DispatchKey::ComplexCPU,
+  DispatchKey::ComplexCUDA,
+  DispatchKey::CustomRNGKeyId,
+  DispatchKey::MkldnnCPU,
+  DispatchKey::SparseCPU,
+  DispatchKey::SparseCUDA,
+  DispatchKey::SparseHIP,
+});
+
+// backend_dispatch_keyset should include all runtime backend keys.
+constexpr DispatchKeySet backend_dispatch_keyset = autogradother_backends | DispatchKeySet({
+  DispatchKey::CPU,
+  DispatchKey::CUDA,
+  DispatchKey::XLA,
+  DispatchKey::PrivateUse1,
+  DispatchKey::PrivateUse2,
+  DispatchKey::PrivateUse3,
+});
+
+// math_dispatch_keyset contains all keys in backend_dispatch_keyset and autograd_dispatch_keyset
+// Alias key DispatchKey::Math maps to math_dispatch_keyset.
+constexpr DispatchKeySet math_dispatch_keyset = backend_dispatch_keyset | autograd_dispatch_keyset;
+
 DispatchKeySet getRuntimeDispatchKeySet(DispatchKey t) {
   TORCH_INTERNAL_ASSERT(t != DispatchKey::Undefined);
   switch (t) {
     case DispatchKey::Autograd:
       return autograd_dispatch_keyset;
-   default:
-     return DispatchKeySet(t);
+    case DispatchKey::Math:
+      return math_dispatch_keyset;
+    default:
+      return DispatchKeySet(t);
+  }
+}
+
+DispatchKeySet getBackendKeySetFromAutograd(DispatchKey t) {
```

Comment:
Could do the same macro-based definition thing for autograd keys as for dispatch keys as a whole (in fact, the macro for the full set could call sub-macros for backend and autograd, I think?)

Again though not worth stopping and doing right away, but if you're back in here for other things it might be worth it.